pkg/webhook/server.go

```diff
@@ -74,14 +92,67 @@ func (s *Server) setDefaults() {
 	s.webhooks = map[string]http.Handler{}
 	s.webhookMux = http.NewServeMux()
 
+	// TODO: probably allow set Port by env.
+	p, err := strconv.Atoi(os.Getenv("WEBHOOKPORT"))
+	if err != nil {
+		log.Error(err, "can't convert WEBHOOKPORT environment variable to int")
+	} else {
+		s.Port = p
+	}
 	if s.Port <= 0 {
 		s.Port = 443
 	}
+
 	if len(s.CertDir) == 0 {
 		s.CertDir = path.Join("/tmp", "k8s-webhook-server", "serving-certs")
 	}
 }
 
+// EnableWebhooksFor registers mutating (defaulting) and validation webhooks for the provided types that
+// implement Defaulter and (or) Validator. Passing in core types doesn't work, since the core types implement
+// neither the Defaulter nor the Validator interfaces.
+// It returns a NotImplementDefaulterValidatorInterfacesError error if a provided object implements
+// neither the Defaulter interface nor the Validator interface.
+func (s *Server) EnableWebhooksFor(objects ...runtime.Object) error {
```

Comment:
I'm not convinced this needs to be a method (as opposed to a helper or helpers).  The helper makes it more composable, easier to test individually, and lets us actually have a type signature that requires an object that implements the interfaces (removing the error path in favor of compile-time checking).